crates/accelerate/src/unitary_synthesis.rs

```diff
@@ -827,11 +829,9 @@ fn preferred_direction(
                                 let cost = match target
                                     .qargs_for_operation_name(decomposer.packed_op.name())
                                 {
-                                    Ok(_) => match target[decomposer.packed_op.name()].get(Some(
-                                        &q_tuple
-                                            .into_iter()
-                                            .collect::<SmallVec<[PhysicalQubit; 2]>>(),
-                                    )) {
+                                    Ok(_) => match target[decomposer.packed_op.name()]
+                                        .get::<TargetQargsRef>(&Some(q_tuple.as_slice()).into())
+                                    {
```

Comment:
Something looks very funky with the typing here.

Reply:
Indeed... Fixed in 9602e7c